integrations/github/oauth.go

```diff
@@ -46,100 +48,98 @@ func (h handler) handleOAuth(w http.ResponseWriter, r *http.Request) {
-	is, _, err := gh.Apps.ListUserInstallations(ctx, &github.ListOptions{})
+	gh, err := newClientFromGitHubAppID(aid)
 	if err != nil {
-		l.Warn("OAuth user token source error", zap.Error(err))
-		c.AbortBadRequest("list installations error")
+		l.Warn("failed to initialize GitHub app client", zap.Error(err))
+		c.AbortBadRequest("failed to initialize GitHub app client")
 		return
 	}
-	foundInstallation := false
-	var i *github.Installation
-	for _, i = range is {
-		if i.ID == nil || *i.ID != id {
-			continue
-		}
-		l.Debug("Verified new GitHub app installation",
-			zap.Stringp("repositorySelection", i.RepositorySelection),
-			zap.Int64p("targetID", i.TargetID),
-			zap.Stringp("targetName", i.Account.Login),
-			zap.Stringp("targetType", i.TargetType),
-		)
-		foundInstallation = true
-		break
+
+	i, _, err := gh.Apps.GetInstallation(r.Context(), iid)
+	if err != nil {
+		l.Warn("failed to get GitHub app installation details", zap.Error(err))
+		c.AbortBadRequest("failed to get GitHub app installation details")
+		return
 	}
-	if !foundInstallation {
-		l.Warn("GitHub app installation details not found")
+
+	if i.ID == nil || *i.ID != iid {
+		l.Warn("GitHub app installation details not found", zap.Any("installation", i))
 		c.AbortBadRequest("GitHub app installation details not found")
 		return
 	}
 
-	if i.AppID == nil || i.ID == nil || i.Account == nil || i.Account.Login == nil {
+	if i.Account == nil || i.Account.Login == nil {
 		l.Warn("GitHub app installation details missing required fields")
 		c.AbortBadRequest("GitHub app installation details missing required fields")
 		return
 	}
 
-	appID := strconv.FormatInt(*i.AppID, 10)
-	installID := strconv.FormatInt(*i.ID, 10)
-	user := string(*i.Account.Login)
+	name := string(*i.Account.Login)
 
-	c.Finalize(sdktypes.NewVars().
-		Set(vars.AuthType, "oauth", false).
-		Set(vars.AppID, appID, false).
-		Set(vars.InstallID, installID, false).
-		Set(vars.InstallKey(appID, installID), user, false))
-}
+	events := fmt.Sprintf("%s", i.Events)
+	events = events[1 : len(events)-1]
 
-func (h handler) tokenSource(ctx context.Context, t *oauth2.Token) oauth2.TokenSource {
-	cfg, _, err := h.oauth.Get(ctx, "github")
+	perms, err := json.Marshal(i.Permissions)
 	if err != nil {
-		return nil
+		perms = []byte(err.Error())
 	}
-	return cfg.TokenSource(ctx, t)
+	ps := strings.ReplaceAll(string(perms[1:len(perms)-1]), `"`, "")
+	ps = strings.ReplaceAll(ps, ",", " ")
+
+	c.Finalize(sdktypes.NewVars().
+		Set(vars.AppID, appID, false).
+		Set(vars.AppName, string(*i.AppSlug), false).
+		Set(vars.InstallID, installID, false).
+		Set(vars.TargetID, strconv.FormatInt(*i.TargetID, 10), false).
+		Set(vars.TargetName, name, false).
+		Set(vars.TargetType, string(*i.TargetType), false).
+		Set(vars.RepoSelection, string(*i.RepositorySelection), false).
+		Set(vars.Permissions, ps, false).
+		Set(vars.Events, events, false).
+		Set(vars.UpdatedAt, i.UpdatedAt.Format(time.RFC3339), false).
```

Comment:
These are being saved but currently not being used anywhere?

Reply:
@daabr likes to keep this stuff around for debugging

Reply:
Exactly - specifically in this case it can help identify and explain issues if an org admin (or another connection init) changes the app's repo selection / approved permissions / selected events.